### PR TITLE
Additional conformance test to assert verbose flag is taken into account

### PR DESCRIPTION
**Pull request by skisel-da**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

**Comment by skisel-da** [comment]
I have noticed that we do not test if `verbose` flag in `GetTransactionsRequest` is not being tested in the conformance test. I was using `ActiveContractsServiceIT` as an example for this test. The idea would be to extend it further in the interface subscription epic for the interfaces.

File: ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceOutputsIT.scala [hunk 1]
```diff
@@ -56,4 +61,77 @@ class TransactionServiceOutputsIT extends LedgerTestSuite {
...
+    for {
+      transactionTree <- transactionTrees
+      eventById <- transactionTree.eventsById
+      (_, tree) = eventById
+      events = tree
+      created = events.getCreated
+    } yield created
+  }
+
+  private def flatTransactionFields(transactions: Seq[Transaction]): Seq[RecordField] =
```
Comment (cocreature) [inline ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceOutputsIT.scala:127]: This function looks like it crashes if there is an archive in the transaction. That’s fine for your test but less fine for a general utility so maybe consider moving this within the test/inline it.

File: ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceOutputsIT.scala [hunk 2]
```diff
@@ -56,4 +61,77 @@ class TransactionServiceOutputsIT extends LedgerTestSuite {
...
+      s"$party expected a contract $text labels, but received $transactions.",
+    )
+    val treeCreateArgs = transactionFields(transactionTreeEvents(transactionTrees))
+    assert(
+      treeCreateArgs.forall(_.label.nonEmpty == labelIsNonEmpty),
+      s"$party expected a contract $text labels, but received $transactionTrees.",
+    )
+  }
+
+  private def transactionTreeEvents(transactionTrees: Seq[TransactionTree]): Seq[CreatedEvent] = {
```
Comment (cocreature) [inline ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceOutputsIT.scala:117]: Similar comment here, this also makes a lot of assumptions about the transaction tree and will crash if they're not satisfied. I think if you want this to be a general utility you should accept arbitrary trees and filter to the create events within.

Alternatively move it wtihin the test where making assumptions about the transaction is perfectly fine.

File: ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceOutputsIT.scala [hunk 3]
```diff
@@ -56,4 +61,77 @@ class TransactionServiceOutputsIT extends LedgerTestSuite {
...
+  }
+
+  private def flatTransactionFields(transactions: Seq[Transaction]): Seq[RecordField] =
+    transactionFields(
+      transactions
+        .flatMap(_.events)
+        .map(_.getCreated)
+    )
+
+  private def transactionFields(createdEvent: Seq[CreatedEvent]): Seq[RecordField] = createdEvent
```
Comment (cocreature) [inline ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceOutputsIT.scala:134]: You could share that with `ActiveContractsServiceIT` but probably small enough that the duplication is easier.

**Review by cocreature — APPROVED** [review]
Tests look good, some stylistic comments.

**Review by remyhaemmerle-da — APPROVED** [review]
The test look fine for me, but if we test for field name erasure we should probably test for type erasure as well (in Record, Variant and Enum)

**Comment by cocreature** [comment]
> The test look fine for me, but if we test for field name erasure we should probably test for type erasure as well (in Record, Variant and Enum)

I'd argue that’s part of unit tests for the enrichment logic. Here we're just testing that the enrichment logic gets triggered. But I don't feel strongly about that.

**Pull request merged**